### ISSUE TITLE
EpicGamesProvider will return discounted non-free games as offers.

### ISSUE THREAD

**Issue by moonstar-x** [moonstar-x/discord-free-games-notifier#14]
#### :bug: Describe the Bug

The EpicGamesProvider will return non-free games that are currently on discount as if they were actual offers.

A check should be made to see if `discountPercentage` is 0 before actually setting it as an offer.

UPDATE: After the whole year that I've hosted this I've never seen this happen until very recently. It seems that the Epic changed how the upcoming offers are being represented in the payload which seems to be conflicting with how the bot checks for the current offers.

**Comment by moonstar-x** [comment]
UPDATE:

Today a false positive offer showed up with the following data:

```json
{
   "title":"Yooka-Laylee and the Impossible Lair",
   "id":"617b64a9631c4df0a66ea8fca480d2b9",
   "namespace":"fa11862b1cfd42e3b3c6395af46efe3e",
   "description":"Yooka-Laylee and the Impossible Lair is a brand-new platform adventure from some of the key creative talent behind 'Donkey Kong Country'. With their arch-nemesis Capital B up to no good the buddy duo needs to spring into action once again to save the day!",
   "effectiveDate":"2019-12-31T13:00:00.000Z",
   "offerType":"BASE_GAME",
   "expiryDate":null,
   "status":"ACTIVE",
   "isCodeRedemptionOnly":false,
   "keyImages":[
      {
         "type":"DieselStoreFrontWide",
         "url":"https://cdn1.epicgames.com/undefined/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S3-1360x766-8faccccfe9ca390483f8a177b7fccbd4.jpg"
      },
      {
         "type":"DieselStoreFrontTall",
         "url":"https://cdn1.epicgames.com/undefined/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S4-510x680-5f1857ce88f7747b88c5121c674470e6.jpg"
      },
      {
         "type":"OfferImageTall",
         "url":"https://cdn1.epicgames.com/undefined/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S4-510x680-5f1857ce88f7747b88c5121c674470e6.jpg"
      },
      {
         "type":"Thumbnail",
         "url":"https://cdn1.epicgames.com/fa11862b1cfd42e3b3c6395af46efe3e/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S5-640x478-54920a9a777ce201c97cd9650d98471a.png"
      },
      {
         "type":"CodeRedemption_340x440",
         "url":"https://cdn1.epicgames.com/fa11862b1cfd42e3b3c6395af46efe3e/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S4-510x680-5f1857ce88f7747b88c5121c674470e6.jpg"
      },
      {
         "type":"OfferImageWide",
         "url":"https://cdn1.epicgames.com/undefined/offer/EGS_PlaytonicGamesLtd_YookaLayleeandtheImpossibleLair_S3-1360x766-8faccccfe9ca390483f8a177b7fccbd4.jpg"
      }
   ],
   "seller":{
      "id":"o-uvtztrtfjdn3xgrwyhbuwwb5z42mbv",
      "name":"Team17 Digital Ltd"
   },
   "productSlug":"yooka-laylee-and-the-impossible-lair/home",
   "urlSlug":"yooka-laylee-and-the-impossible-lair",
   "url":null,
   "items":[
      {
         "id":"fd71bd8498794b1d9efe0d0b9ce2fc52",
         "namespace":"fa11862b1cfd42e3b3c6395af46efe3e"
      }
   ],
   "customAttributes":[
      {
         "key":"com.epicgames.app.blacklist",
         "value":"[]"
      },
      {
         "key":"publisherName",
         "value":"Team17 Digital Ltd"
      },
      {
         "key":"developerName",
         "value":"Playtonic Games Ltd."
      },
      {
         "key":"com.epicgames.app.productSlug",
         "value":"yooka-laylee-and-the-impossible-lair/home"
      }
   ],
   "categories":[
      {
         "path":"freegames"
      },
      {
         "path":"games"
      },
      {
         "path":"games/edition/base"
      },
      {
         "path":"games/edition"
      },
      {
         "path":"applications"
      }
   ],
   "tags":[
      {
         "id":"14944"
      },
      {
         "id":"18777"
      },
      {
         "id":"1370"
      },
      {
         "id":"14346"
      },
      {
         "id":"9547"
      },
      {
         "id":"16011"
      },
      {
         "id":"1117"
      },
      {
         "id":"9549"
      },
      {
         "id":"1151"
      },
      {
         "id":"15375"
      }
   ],
   "catalogNs":{
      "mappings":[
         {
            "pageSlug":"yooka-laylee-and-the-impossible-lair",
            "pageType":"productHome"
         }
      ]
   },
   "offerMappings":[
      
   ],
   "price":{
      "totalPrice":{
         "discountPrice":749,
         "originalPrice":2999,
         "voucherDiscount":0,
         "discount":2250,
         "currencyCode":"USD",
         "currencyInfo":{
            "decimals":2
         },
         "fmtPrice":{
            "originalPrice":"$29.99",
            "discountPrice":"$7.49",
            "intermediatePrice":"$7.49"
         }
      },
      "lineOffers":[
         {
            "appliedRules":[
               {
                  "id":"bb0fe7711f5c43dcaec4cca852353e86",
                  "endDate":"2022-02-10T16:00:00.000Z",
                  "discountSetting":{
                     "discountType":"PERCENTAGE"
                  }
               }
            ]
         }
      ]
   },
   "promotions":{
      "promotionalOffers":[
         {
            "promotionalOffers":[
               {
                  "startDate":"2022-01-27T16:00:00.000Z",
                  "endDate":"2022-02-10T16:00:00.000Z",
                  "discountSetting":{
                     "discountType":"PERCENTAGE",
                     "discountPercentage":25
                  }
               }
            ]
         }
      ],
      "upcomingPromotionalOffers":[
         {
            "promotionalOffers":[
               {
                  "startDate":"2022-02-03T16:00:00.000Z",
                  "endDate":"2022-02-10T16:00:00.000Z",
                  "discountSetting":{
                     "discountType":"PERCENTAGE",
                     "discountPercentage":0
                  }
               }
            ]
         }
      ]
   }
}
```

For a valid offer it would seem that `discountPercentage` is 0 but maybe a check on `startDate` and `endDate` should also be done.

**Comment by moonstar-x** [comment]
Fixed in #22
